services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/CommandRouterServiceImpl.java

```diff
@@ -247,14 +250,14 @@ private void processTenantQueue(final Set<String> processedTenants, final Span p
                         if (t instanceof ServerErrorException) {
```

Comment:
I'm still not sure whether there aren't kinds of ServerErrorExceptions here where we _don't_ want to retry adding the command consumer indefinitely.

I guess most scenarios would be fine. E.g. also for a "amqp:unauthorized-access - link disallowed by local policy" or "amqp:not-found - Node not found" failure when trying to open the command consumer receiver link, ClientServiceExceptions are thrown.

But I think it is a bit risky to make the assumption here that any kind of ServerErrorExceptions (e.g. status 500) is resolvable and stops reoccurring here at some point.
When using Kafka for example, there are no replies to the consumer creation request like "amqp:unauthorized-access" which we would map to a ClientServiceException.
If a tenant-specific Kafka topic has been deleted at some point while the tenant still exists, creation of the Kafka-based command consumer will fail with an UNAVAILABLE ServerErrorException. To keep retrying here, checking for the existence of the topic would cost unnecessary Kafka server requests.

Reply:
Do you have a suggestion for a better approach?

Reply:
I would limit the number of retries per specific tenant overall.
And I would keep a list of tenants for which processing failed and trigger the retry-operation concerning these tenants after a certain delay.
See https://github.com/eclipse/hono/issues/2705#issuecomment-855740463.

Reply:
Ok, good.

Reply:
I have pushed corresponding changes ...